app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,31 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_cart_updates_form # rubocop:disable MethodLength, AbcSize
+    flash.clear
+    begin
+      # set the start and end date to updated values
+      # converts the dates into the expected format
+      start_date = Date.strptime(params[:form][:start_date], '%m/%d/%Y')
```

Comment:
Right now, the .toDate method used by update_cart doesn't handle dates specified in the format MM/DD/YY which is what the date picking calendar returns. On the actual cart there are hidden fields which if I recall correctly are doing the same sort of string manipulation logic as this in the JavaScript and storing the modified values in the hidden fields. I thought that moving the string manipulation code into the controller was clearer than storing hidden fields with manipulated values.